src/function_data.jl

```diff
@@ -254,27 +219,38 @@ deserialize(::Type{FunctionData}, val::Dict) =
 Get a bare numerical representation of the data represented by the FunctionData
 """
 function get_raw_data end
-get_raw_data(fd::LinearFunctionData) = get_proportional_term(fd)
+get_raw_data(fd::LinearFunctionData) = (get_proportional_term(fd), get_constant_term(fd))
```

Comment:
get_raw_data here is not type stable. Test it with a type annotation inside a function using @code_warntype

Reply:
Indeed — here's a MWE, for the record:
```julia
@code_warntype IS.transform_array_for_hdf(SortedDict(DateTime("2023-01-01") => [LinearFunctionData(1.0)]))
```
I'll work on fixing this.

Reply:
Open an issue to tackle this so we can merge this + the PSY companion to continue on to the PSI work.

Reply:
Done, https://github.com/NREL-Sienna/InfrastructureSystems.jl/issues/352